utt/wallet-cli/include/fileBasedUserStorage.hpp

```diff
@@ -0,0 +1,50 @@
+// UTT Client API
+//
+// Copyright (c) 2020-2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <utt-client-api/IStorage.hpp>
+#include <mutex>
+#include <optional>
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+namespace utt::client {
+class FileBasedUserStorage : public IStorage {
+ public:
+  FileBasedUserStorage(const std::string& path);
+  bool isNewStorage() override;
+  void setKeyPair(const std::pair<std::string, std::string>&) override;
+  void setLastExecutedSn(uint64_t) override;
+  void setClientSideSecret(const libutt::api::types::CurvePoint&) override;
+  void setSystemSideSecret(const libutt::api::types::CurvePoint&) override;
+  void setRcmSignature(const libutt::api::types::Signature&) override;
+  void setCoin(const libutt::api::Coin&) override;
+  void removeCoin(const libutt::api::Coin&) override;
+  void startTransaction() override;
+  void commit() override;
```

Comment:
IMHO, non-business logic methods, such as `isNewStorage`, `startTransaction`, `commit` should be in a separate interface.

Reply:
Let's keep it a single interface - ignore this comment for now.